identity-apps-core/apps/authentication-portal/src/main/webapp/identifierauth.jsp

```diff
@@ -104,8 +104,8 @@
 %>
 
 <%
-    String clientId = request.getParameter("client_id");
-    String sp = request.getParameter("sp");
+    String clientId = Encode.forUriComponent(request.getParameter("client_id"));
+    String sp = Encode.forUriComponent(request.getParameter("sp"));
```

Comment:
These places also we use `encode.forjavascriptblock` or suitable encoding method depending on the usage.